backend/src/main/java/ee/tlu/forum/service/ThreadService.java

```diff
@@ -72,14 +75,17 @@ public void deleteThreadById(Long id) {
     }
 
     @Override
-    public Thread editThread(Thread thread) {
+    public Thread editThread(Thread thread, String token) {
         if (thread.getId() == null) {
             throw new BadRequestException("Field must contain a thread ID");
         }
         Optional<Thread> threadOptional = threadRepository.findById(thread.getId());
         if (threadOptional.isEmpty()) {
             throw new NotFoundException("Thread with ID " + thread.getId() + " does not exist");
         }
+
+        tokenHelper.hasRoleOrUsername(token, threadOptional.get().getAuthor().getUsername(), "ROLE_ADMIN");
```

Comment:
Would this stop the author from editing their thread?

Reply:
No, it stops other users from editing other people's threads